src/backings/structural/array.ts

```diff
@@ -88,7 +88,7 @@ export class BasicArrayStructuralHandler<T extends ArrayLike<unknown>> extends S
   }
 }
 
-export class CompositeArrayStructuralHandler<T extends ArrayLike<object>> extends StructuralHandler<T> {
+export class CompositeArrayStructuralHandler<T extends ArrayLike<ObjectLike>> extends StructuralHandler<T> {
```

Comment:
If I put CompositeValue, that allows array element to be another array, which is perfectly fine. But our types doesn't reflect that anywhere so there is huge number of type errors. It's kinda hard because wherever I change type I cause chain reaction.

Not sure how to proceed. Maybe we can in separate PRs fix types from bottom up?